src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.Unix.cs

```diff
@@ -2,10 +2,25 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Text;
+
 namespace System.Runtime.InteropServices
 {
     public static partial class Marshal
     {
         private static bool IsWin32Atom(IntPtr ptr) => false;
+
+        internal static unsafe int StringToAnsiString(string s, byte* buffer, int wideCharLen, bool bestFit = false, bool throwOnUnmappableChar = false)
+        {
+            int convertedBytes;
+
+            fixed (char* pChar = s)
+            {
+                convertedBytes = Encoding.UTF8.GetBytes(pChar, s.Length, buffer, wideCharLen);
+                buffer[convertedBytes] = 0;
```

Comment:
Is buffer guaranteed to be at least convertedBytes+1 long?

Reply:
(This can also be moved outside of the fixed.)

Reply:
>Is buffer guaranteed to be at least convertedBytes+1 long?

Yes.

Reply:
Can we pass in the buffer length and assert it?

Reply:
It looks like we already assert it in the Windows code; we should do the same in the Unix code.

Reply:
ok, copied the same assert from windows version

Reply:
Could you please also rename the `wideCharLen` argument to better capture the meaning, e.g. `bufferLength` ?